library/oeUI/src/OemrUI.php

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /**
+    * Used to create the html string that will display the formatted expandable icon - fa-expand or fa-compress.
+    *
+    * @param array $arrExpandable - int|bool - the current state of the form and int|bool - whether form is expandable or not
+    *
+    * @return array containing string $expandable_icon - the formatted html string of the expand icon and string $container - the value of the container class 'container' or 'container-fluid'
+    *
+    */
+    //private function expandIcon($expandable = '', $current_state = '')
+    private function expandIcon()
+    {
+        $current_state = $this->current_state;
+        $expandable = $this->expandable;
+        
+        if ($current_state) {
+            $container = 'container-fluid';
+            $expand_title = xl('Click to Contract and set to henceforth open in Centered mode');
+            $expand_icon_class = 'fa-compress oe-center';
+        } else {
+            $container = 'container';
+            $expand_title = xl('Click to Expand and set to henceforth open in Expanded mode');
+            $expand_icon_class = 'fa-expand oe-expand';
+        }
+        if ($expandable) {
+            $expandable_icon = "<i id='exp_cont_icon' class='oe-superscript-small expand_contract fa " .  attr($expand_icon_class) . "'" . " title='" . attr($expand_title) . "' 
+            aria-hidden='true'></i>";
+        }
+        return array($expandable_icon, $container);
+    }
+    
+    /**
+    * Will return the container class value either 'container' or 'container-fluid'
+    *
+    * @return string $container that will reflect the current state of the page i.e. expanded = 'container-fluid' or centered = 'container'
+    */
+    public function oeContainer()
+    {
+        $arrexpandIcon = $this->expandIcon();
+        $container = $arrexpandIcon[1] ? $arrexpandIcon[1]:'container';
+        return $container;
+    }
+    
+    /**
+    * Used to create the html string that will display the formatted action/re-direction icon - for conceal, reveal, search, reset, link and back.
+    *
+    * @param array $arrAction has 3 elements - string - type of action, string - optional title to be used in tooltip
+    * and string - the file name or url to be redirected to, only the 3 re-directions reset, link or back need a href value
+    * the 3 actions conceal, reveal, search will only use the default title strings
+    *
+    * @return string $action_icon that will output the action icon html string
+    *
+    */
+    private function actionIcon($arrAction = array())
+    {
+        $arrAction = $this->arrAction;
+        if ($arrAction) {
+            $action = $arrAction[0];
+            $action_title = $arrAction[1];
+            $action_href = $arrAction[2];
+        }
+        $action_href = ($action_href) ? $action_href : "#";
+        switch ($action) {
+            case "reset":
+                $action_title = ($action_title) ? $action_title : "Reset";
```

Comment:
Translate the Reset with xl() function

Reply:
Done here :)
`$action_icon = "<a href='" . attr($action_href) ."' onclick='top.restoreSession()'><i id='advanced-action' class='fa fa-undo fa-2x small' title='" . xla($action_title) ."' aria-hidden='true'></i></a>";`

Reply:
There are generally 2 forces at work on the translation and escaping:
1. For translations, it is much better to translate as upstream as possible, so the constants get captured in the scripts that grab them from the codebase.
2. For html escaping, best to escape as downstream as possible.
Which is why should translate the Reset on the same line (or when it is entered into the class call), but do the escape call when it is outputted to html.